src/tokenizer.c

```diff
@@ -0,0 +1,503 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   tokenizer.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: chuhlig <chuhlig@student.42.fr>            +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/06/28 20:55:50 by chuhlig           #+#    #+#             */
+/*   Updated: 2024/08/04 14:32:56 by chuhlig          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+#include "token.h"
+
+char	*ft_strncpy(char *s1, char *s2, int n)
+{
+	int	i;
+
+	i = -1;
+	while (++i < n && s2[i])
+		s1[i] = s2[i];
+	return (s1);
+}
+
+void	print_token(t_token *token)
+{
+	if (token->type == STRING_TOKEN)
+	{
+		printf("STRING_TOKEN: %s\n", token->content.string);
+	}
+	else if (token->type == REDIR_TOKEN)
+	{
+		printf("REDIR_TOKEN: %d\n", token->content.redir_type);
+	}
+	else if (token->type == PIPE_TOKEN)
+	{
+		printf("PIPE_TOKEN\n");
+	}
+	else if (token->type == NEWLINE_TOKEN)
+	{
+		printf("NEWLINE_TOKEN\n");
+	}
+}
+
+void	conditional_print(char *string, int start_of_string, int i,
+	t_token **token_list)
+{
+	char	*line;
+	int		len;
+
+	len = i - start_of_string + 1;
+	if (len > 0)
+	{
+		line = (char *)malloc(len);
+		if (!line)
+		{
+			exit(EXIT_FAILURE);
+		}
+		ft_strncpy(line, string + start_of_string, len);
+		line[len] = '\0';
```

Comment:
fsanitizer shows a buffer overflow here. This should be easily fixable by either changing this line to
```suggestion
		line[len - 1] = '\0';
```
or line 55 to
```
line = (char *)malloc(len + 1);
```

Reply:
done